x/metadata/keeper/keeper_test.go

```diff
@@ -388,69 +505,84 @@ func (s *KeeperTestSuite) TestValidateAllOwnersAreSigners() {
 			msgTypeURL: "",
 			errorMsg:   "",
 		},
+	}
+
+	for n, tc := range tests {
+		s.T().Run(n, func(t *testing.T) {
+			err := s.app.MetadataKeeper.ValidateAllOwnersAreSignersWithAuthz(s.ctx, tc.owners, tc.signers, tc.msgTypeURL)
+			if len(tc.errorMsg) == 0 {
+				assert.NoError(t, err, "ValidateAllOwnersAreSigners unexpected error")
+			} else {
+				assert.EqualError(t, err, tc.errorMsg, "ValidateAllOwnersAreSigners error")
+			}
+		})
+	}
+}
+
+func (s *KeeperTestSuite) TestValidateAllOwnersAreSignersWithGenericAuthorization() {
+
+	tests := map[string]struct {
+		owners     []string
+		signers    []string
+		msgTypeURL string
```

Comment:
Here too. Can we put the `TestValidateAllOwnersAreSigners` tests back to the way the were before?